src/solver/toolkit.c

```diff
@@ -412,6 +412,75 @@ EXPORT_TOOLKIT int  swmm_getSimulationParam(SM_SimSetting type, double *value)
     return error_code;
 }
 
+EXPORT_TOOLKIT int DLLEXPORT swmm_hotstart(SM_HotStart type, const char *hsfile)
+///
+/// Input:   type = Hotstart option USE/SAVE (SM_HotStart)
+///          hotstart = file ID name (able to overwrite)
+/// Return   API Error
+/// Purpose: Allows selecting a HSF to use or save
+{
+    int error_code = 0;
+    // Check if Open
+    if(swmm_IsOpenFlag() == FALSE)
+    {
+        error_code = ERR_TKAPI_INPUTNOTOPEN;
+    }
+    // Check if Simulation is Started
+    else if (swmm_IsStartedFlag() == TRUE && type == SM_HOTSTART_USE)
+    {
+        error_code = ERR_TKAPI_SIM_RUNNING;
+    }
+    // Check if Simulation is NOT started
+    else if (swmm_IsStartedFlag() == FALSE && type == SM_HOTSTART_SAVE)
+    {
+        error_code = ERR_TKAPI_SIM_NRUNNING;
+    }
+    else
+    {
+        switch(type)
+        {
+            case SM_HOTSTART_USE:
+            {
+                // Fhotstart1 is the file to to be read
+                char fl_name[MAXFNAME];
+                sstrncpy(fl_name, hsfile, MAXFNAME);
+                Fhotstart1.mode = USE_FILE;
+                sstrncpy(Fhotstart1.name, addAbsolutePath(fl_name), MAXFNAME);
```

Comment:
* When saving a hotstart,  `addAbsolutePath` is not used...so the hotstart path is relative to the current working directory (CWD).

* When using a hotstart, `addAbsolutePath` **IS**  used...so the hotstart path becomes relative to the inp file.

This discontinuity might be confusing to users who run a pyswmm model using an input file that is not in the CWD, you'd need to either always use an absolute path to reference the HSF, or change how to reference the path to the HSF depending on whether your SAVING or USING.

I think when using the API, it makes more prescribe paths relative to the CWD, so I'd consider removing `addAbsolutePath`.